### DISCUSSION TITLE
Allow Maintain role to modify branch protection checks but not approver count

### DISCUSSION THREAD

**Discussion by LamarrD** [community/community#37147] [Repositories]
In our organization we allow teams to be maintainers of their own repositories. This is done to prevent anyone from removing the branch protection requirement of at least 1 approving review. But, this also prevents teams from being able to add/modify their own required checks.

We would like the `Maintain` role to be able to modify required checks. Alternatively even if this permission was made more granular so that we could have `modify branch protection approvals`: `False` and `modify branch protection checks`: `True` that would also be great.

**Comment by ossanna16** [comment]
Hi there @LamarrD and welcome to our community! Thank you for asking a great question 🙂

**Reply by LamarrD** [reply]
Hi @ossanna16, do you know when I'd get an answer on if this is possible or planned in the future?

**Reply by ossanna16** [reply]
Hi @LamarrD! Unfortunately I can't provide any further insights on this but our Product Teams are keeping a close eye on the Community on Discussions and appreciate hearing your feedback about repositories. It helps us improve our product.

**Comment by ewega** [comment]
Hi @LamarrD ,

Responding here for you and the community. Some info dump first:

**Roles & Access**
Any team/members with `Write` access in a repo can create status checks. Managing branch protection rules requires `Admin` access. In your scenario you gave your teams `Maintain` access so they technically can create checks because `Maintain` > `Write` (e.g. security tests for PRs using a GitHub Actions workflow), but could not modify required check settings (i.e. does this specific test need to pass for a PR to be mergeable).

Look through the individual permissions and the roles they're mapped to for a repo [here](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization).

**Custom Roles**
There exists [Custom Repository Roles](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-peoples-access-to-your-organization-with-roles/managing-custom-repository-roles-for-an-organization). You can create custom roles at the org-level, and then have them assigned to certain repos. With these custom roles, you can provide fine-grained permissions.

**Your Use-Case**
Note that `Admin` access for a repository is required to interact with branch protection rules. You cannot add a permission to `Maintain` to interact with branch protection rules currently. What I would suggest in your use case without too much context is providing specific team _managers/leaders_ with `Admin` access to repositories so they can fully manage approvals and required checks. The rest of the team can have `Maintain` access so they can create status checks but not manage branch protection rules.

This is the flow I've typically seen. Curious your thoughts on it. If this doesn't work for you, do you feel you would like custom roles to be able to perform admin-level activities?
